src/uct/base/uct_md.c

```diff
@@ -368,17 +368,74 @@ static ucs_status_t uct_mem_check_flags(unsigned flags)
     return UCS_OK;
 }
 
-ucs_status_t uct_md_mem_alloc(uct_md_h md, size_t *length_p, void **address_p,
-                              unsigned flags, const char *alloc_name, uct_mem_h *memh_p)
+ucs_status_t uct_mem_alloc_check_params(const uct_mem_alloc_params_t *params)
 {
+    unsigned flags;
     ucs_status_t status;
 
-    status = uct_mem_check_flags(flags);
-    if (status != UCS_OK) {
-        return status;
+    if (params->field_mask & UCT_MEM_ALLOC_PARAM_FIELD_FLAGS) {
+        status = uct_mem_check_flags(params->flags);
+        if (status != UCS_OK) {
+            return status;
+        }
+    }
+
+    flags = params->flags;
```

Comment:
what if no flags?

Reply:
You mean user sets PARAM_FIELD_FLAGS but doesn't set flags or sets flags to zero?

Reply:
there is no check for field_mask here

Reply:
The change here https://github.com/openucx/ucx/pull/5562/commits/add6339434cc29e5ee635c89e5bba5d79fc9d8f8#diff-6282d9e471573ebace2884f6debada46R383 in commit add6339 should address the concern right?